lib/matplotlib/axis.py

```diff
@@ -1838,6 +1841,9 @@ def set_ticks(self, ticks, labels=None, *, minor=False, **kwargs):
         choice to prevent the surprise of a non-visible tick. If you need
         other limits, you should set the limits explicitly after setting the
         ticks.
+
+        Do not mix automatically and manually placed ticks with this method.
+        See :ghissue:`22262` for discussion.
```

Comment:
Can try you write this down more explicitly? If we write down extra comment they must be easily understandable for the user. Without having looked at #22262, this doesn't make sense to me. I either have a `Locator` that automatically places ticks, or I have set tick positions explicitly, which results in a `FixedLocator`. How can I mix both? Generally, I'm 👎 on referring to GitHub issues for explanation. They are a historic record of a discussion but not a condensed and easy to understand description.

Reply:
The problem with mixing auto and manual ticks comes from a simple and very naive idea of how to add a special tick to an existing axis. Essentially getting and appending to the list of ticks already on the axis:

```python
ticks = axes.get_yticks()
ticks_added = np.append(ticks, [1.234])
axes.set_yticks(ticks_added)
```

For an outside user, there is nothing to suggest that this shouldn't be done and in itself it doesn't look that unreasonable to do. But it turns out to be an edge case where the underlying implementation leaks through.

Reply:
I can try to reformulate this more explicitly in words, but I think it can't really get any better than the code. But then again, that is a bit verbose.

My intent is to warn any future users that could fall into this same trap when trying to approach the task in this deceptively simple way.

Reply:
There's more, actually, not limited to the edge case. The edge case just results in the `Locator` changing unilaterally without the `Formatter` changing to match it, as is normally the case when the plot is made, and extra ticks appearing in places not expected by the `Formatter` (the "gentlemen's agreement" is broken).

But in this case, even if all ticks are set manually, this will pop up whenever a tick is set to something other than what the `Formatter` is trained to expect from the default `Locator` for the type of plot.

The warning to not mix auto and manual ticks comes from the issue discussion, but the problem of unlabeled ticks is not directly related to the mixing itself, if one thinks about it carefully.

So I guess this last sentence can be deleted if it is made clear enough that the problem lies with the `Formatter` alone.

Reply:
> So I guess this last sentence can be deleted if it is made clear enough that the problem lies with the Formatter alone.

Do you think the above description makes it clear enough?

Reply:
Yes, I believe so.

I have further tested to make sure that even this apparently discouraged *mixing* will work, so there is no need for this sentence.

The test:
```python
xs = np.linspace(0, L, 101)
ys = 0.05 / np.abs(np.exp(-xs))

fig = plt.figure()
axes = fig.subplots()
axes.semilogy(xs, ys)

sticx = np.append(axes.get_yticks(), [0.1, 0.5, 1, 2, 4, 10])
sticlbl = [f"{tick:g}" for tick in sticx]
axes.set_yticks(sticx, labels=sticlbl)
```
gives
![Figure 2022-01-21 113522](https://user-images.githubusercontent.com/6322902/150512510-c9369991-c149-4b17-ac8a-a8aecc11859a.png)
which is what we wanted.

Reply:
Test of the same "naive" method on both axes with `loglog` plot:
```python
xs = np.linspace(0, L, 101)
ys = 0.05 / np.abs(np.exp(-xs))

fig = plt.figure()
axes = fig.subplots()
axes.loglog(xs, ys)

sticx = np.append(axes.get_xticks(), [0.1, 0.25, 0.5, 1, 2, 4, 6])
sticxlbl = [f"{tick:g}" for tick in sticx]
sticy = np.append(axes.get_yticks(), [0.1, 0.5, 1, 2, 4, 10])
sticylbl = [f"{tick:g}" for tick in sticy]
axes.set_xticks(sticx, labels=sticxlbl)
axes.set_yticks(sticy, labels=sticylbl)
```
gives
![Figure 2022-01-21 114438](https://user-images.githubusercontent.com/6322902/150513827-228b4c88-78c0-43f2-a8e3-a4655b60cce6.png)